src/v/cluster/archival/purger.cc

```diff
@@ -225,7 +225,23 @@ purger::collect_manifest_paths(
             continue;
         }
 
-        collected.spillover.push_back(std::move(item.key));
+        // The spillover manifest path is of the form
+        // "{prefix}/{manifest.bin().x.x.x.x.x.x}" Find the index of the last
+        // '/' in the path, so we can check just the filename (starting from the
+        // first character after '/').
+        const size_t filename_idx = path.rfind('/');
+        if (filename_idx == std::string_view::npos) {
+            continue;
+        }
+
+        // File should start with "manifest.bin()", but it should have
+        // additional spillover components as well.
+        std::string_view file = path.substr(filename_idx + 1);
+        if (
+          file.starts_with(cloud_storage::partition_manifest::filename())
+          && !file.ends_with(cloud_storage::partition_manifest::filename())) {
```

Comment:
Why is `!file.ends_with(cloud_storage::partition_manifest::filename())` needed? Can we make this less "hacky" by checking a property of the list result elements? Anything that tells us that an entry is directory vs file?

E.g. the blob properties in the list call contains `        <ResourceType>file | directory</ResourceType>
`

There is also seem to be support for a query param `showonly={deleted,files,directories}`

```
-files: Optional. Version 2020-12-06 and later. Only for accounts enabled with hierarchical namespace. When a request includes this parameter, the list only contains files.
```

Reply:
We could potentially change the original call to `_api.list_objects()` by passing in an `item_filter` that essentially uses the existing checks as a predicate.

Also, we don't actually parse the `ResourceType` out in our `xml_sax_parser` for ` list_bucket_item` implementation currently.